### PR TITLE
Manage FMU simulation times

### PR DESCRIPTION
**Pull request by Claire-Eleutheriane**
Enable the user to set start and final time for the FMU simulation time in *FMUPointToFieldFunction* and *OpenTURNSFMUPointToFieldFunction*

Raise an error if the mesh start or stop time are outside the simulation start and stop time.

### CODE REVIEW

File: otfmi/otfmi.py [hunk 1]
```diff
@@ -443,16 +450,21 @@ class OpenTURNSFMUPointToFieldFunction(ot.OpenTURNSPythonPointToFieldFunction):
...
                  inputs=None, outputs=None,
                  initialization_script=None, kind=None,
-                 expect_trajectory=False, **kwargs):
+                 expect_trajectory=False, start_time=None,
+                 final_time=None, **kwargs):
         self.load_fmu(path_fmu=path_fmu, kind=kind)
 
         self._set_inputs_fmu(inputs_fmu)
         self._set_outputs_fmu(outputs_fmu)
+        self.mesh = mesh
```
Comment (jschueller) [inline otfmi/otfmi.py:459]: pas besoin de doublon: le mesh peut être récupéré par la classe mère via getOutputMesh()

File: otfmi/otfmi.py [hunk 2]
```diff
@@ -541,6 +553,47 @@ def _set_outputs(self, outputs=None):
...
+        if start_time is not None:
+            self.start_time = start_time
+        else:
+            self.start_time = self.model.get_default_experiment_start_time()
+
+    def _assert_mesh_pertinence(self):
+        """Raise an error if the mesh is not comprised between the start and
+        final simulation time.
+        """
+        mesh_min = self.mesh.getValues()[0]
```
Comment (jschueller) [inline otfmi/otfmi.py:586]: I expected something like getOutputMesh().getVertices().getMin()[0]
Funny thing is that you can specify a mesh that travels back through time

File: otfmi/otfmi.py [hunk 3]
```diff
@@ -541,6 +553,47 @@ def _set_outputs(self, outputs=None):
...
+        assert mesh_min >= self.start_time, """The mesh start time must be >= to FMU start time.\n
+            To set the FMU start time, use the argument *start_time* in
+            FMUPointToFieldFunction constructor."""
+
+        mesh_max = self.mesh.getValues()[-1]
```
Comment (jschueller) [inline otfmi/otfmi.py:591]: same

File: test/test_OpenTURNSFMUPointToFieldFunction.py [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+        """Check if incoherent final time raises an error.
+        """
+        model_fmu = otfmi.OpenTURNSFMUPointToFieldFunction(
+            self.mesh,
+            self.path_fmu,
+            inputs_fmu=['infection_rate', 'healing_rate'],
+            outputs_fmu=['infected'],
+            final_time=10)
+        self.assertRaises(AssertionError, model_fmu._exec, input_value)
+
```
Comment (jschueller) [inline test/test_OpenTURNSFMUPointToFieldFunction.py:98]: I had a test that actually tested the value of the function at final time, I see it disappeared

**Comment by jschueller** [comment]
on pourrait aussi vouloir  cette fonctionnalité pour la version vectorielle (FMUFunction)

File: test/test_OpenTURNSFMUPointToFieldFunction.py (hunk 4 above)
Comment (Claire-Eleutheriane) [reply test/test_OpenTURNSFMUPointToFieldFunction.py:98]: Indeed. 
Before, the "epid" model was written in the Python file, then exported as FMU and compared to a manually set value.
Where did this value come from ? I didn't know.

What I changed: the epid.fmu pre-exists among OTFMI examples.
The output value from OTFMI' simulation is compared to the output from PyFMI simulation. As OTFMI interpolation modifies the exact values, the check verifies if OTFMI's output is comprised between PyFMI's 5 last values.

File: otfmi/otfmi.py (hunk 1 above)
Comment (Claire-Eleutheriane) [reply otfmi/otfmi.py:459]: Ok

File: otfmi/otfmi.py (hunk 2 above)
Comment (Claire-Eleutheriane) [reply otfmi/otfmi.py:586]: Ok

File: otfmi/otfmi.py (hunk 3 above)
Comment (Claire-Eleutheriane) [reply otfmi/otfmi.py:591]: ok

**Pull request merged**